src/events/InputEventGenerator.cc

```diff
@@ -83,7 +83,23 @@ void InputEventGenerator::poll()
 	auto* curr = &event2;
 	bool pending = false;
 
-	while (SDL_PollEvent(curr)) {
+	// For the first event, optionally wait with timeout. This allows
+	// efficient blocking when the emulator is paused, waking immediately
+	// on any SDL event (mouse, keyboard, etc.) or on timeout.
+	auto getNextEvent = [&, firstEvent = true]() mutable -> bool {
+		if (firstEvent && timeoutMs) {
+			firstEvent = false;
+			return SDL_WaitEventTimeout(curr, *timeoutMs) == 1;
+		}
+		return SDL_PollEvent(curr) == 1;
+	};
```

Comment:
Can we simplify this to something like:
```
auto getNextEvent = [&]{
    if (timeoutMs) {
        auto t = *timeoutMs;
        timeoutMs = {};
        return SDL_WaitEventTimeout(curr, t);
    }
    return SDL_PollEvent(curr);
};
```
In other words: reuse the `timeoutMs` parameter instead of introducing a new variable `firstEvent`.
Just an idea, and untested.

Or even:
```
auto getNextEvent = [&]{
    return timeoutMs ? SDL_WaitEventTimeout(curr, *std::exchange(timeoutMs, {}))
                     : SDL_PollEvent(curr);
};
```